src/components/Button.tsx

```diff
@@ -0,0 +1,27 @@
+import cn from 'classnames';
+import { MouseEventHandler, ReactNode } from 'react';
+
+type Button = {
+  type: "button" | "submit" | "reset" | undefined;
```

Comment:
The `undefined` here could be replaced with an optional field.

```suggestion
  type?: "button" | "submit" | "reset";
```

Also, its important we have a default behavior. If no `type` is provided then lets use `'button'` bu default.